tests/migration/sle12_online_migration/zypper_patch.pm

```diff
@@ -28,6 +28,7 @@ sub run {
     fully_patch_system;
     remove_ltss;
     power_action('reboot', keepconsole => 1, textmode => 1);
+    reconnect_mgmt_console if (check_var('ARCH', 'ppc64le'));
```

Comment:
The architecture does not need this. you want to look for backend here only. You should be able to just call `reconnect_mgmt_console` without any condition, same as done in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/boot/reconnect_mgmt_console.pm#L20 as the implementation of `reconnect_mgmt_console` is already checking for specifics of architecture and backend. If this creates problems on other architecture/backend then most likely there is another workaround interfering which should be streamlined.

Reply:
As before we don't have reconnect_mgmt_console at here. All platform cases can pass. I think we'd better add this check_var('ARCH', 'ppc64le') at here. Or it will impact others cases. I need rerun all of other three platforms cases to check if it can work or not

Reply:
The code of `reconnect_mgmt_console` same as the test API as well as many helper functions is designed to provide a backend-agnostic behaviour to the outside. So please do only introduce the condition if it is really necessary not as optimization that is not based on reliable data. There is no significant runtime overhead for calling a function that in the end does nothing but on the other hand code maintenance by humans is a cost worth to consider. Might be an interesting for the topic: https://stackify.com/premature-optimization-evil/ . As we work on test code itself the approach should be to use the generic functions where possible and whenever a problem is encountered, e.g. a test would fail on other backends then handle them accordingly, at best already within the lib-function.